packages/flutter/lib/src/services/text_input.dart

```diff
@@ -723,6 +723,13 @@ TextAffinity? _toTextAffinity(String? affinity) {
 
 /// A floating cursor state the user has induced by force pressing an iOS
 /// keyboard.
+///
+/// ## Floating cursor
+///
+/// The floating cursor is an iOS feature in which long pressing and dragging on
+/// the spacebar of the soft keyboard causes the cursor in the text field to
+/// move along with the drag gesture. It's used to precisely position the cursor
+/// in some editable text.
```

Comment:
that doesn't really help me understand what it is, or why it has multiple colors, etc. would it be possible to describe it in more detail, or maybe generate screenshots showing it (in assets-for-api-docs) and pointing to those?

The audience here is people who have never used iOS. They probably need to know all the various modes it can be in, the various properties that can be used to configure it, and so on.

Reply:
(the writing prompt i usually give myself is, "if someone is looking at this, that means they have a question that they're trying to answer. what is the question? what will follow-up questions be?")

Reply:
Updated, let me know what you think.